data-science-onramp/ai-platform/sklearn_test.py

```diff
@@ -0,0 +1,75 @@
+import glob
+import os
+import tarfile
+import time
+import uuid
+
+from google.cloud import storage
+from googleapiclient import discovery
+import pytest
+
+STAGING_BUCKET = f"sklearn-job-dir-{uuid.uuid4()}"
+INPUT_DIR = "inputs"
+TRAINER_DIR = "modules"
+MODELS_DIR = "models"
+JOB_DIR = f"gs://{STAGING_BUCKET}/{MODELS_DIR}"
+TRAIN_DATA = "train.csv"
+TRAINER_TAR = "trainer.tar.gz"
+TRAIN_DATA_PATH = f"gs://{STAGING_BUCKET}/{INPUT_DIR}/{TRAIN_DATA}"
+
+PROJECT_ID = os.environ["GOOGLE_CLOUD_PROJECT"]
+REGION = "us-central1"
+MODEL_NAME = f"sklearn-test-{uuid.uuid4()}"
+JOB_ID = f"sklearn_{str(uuid.uuid4())[:7]}"
+
+TERMINAL_STATES = ["SUCCEEDED", "FAILED", "CANCELLING", "CANCELLED"]
+
+@pytest.fixture(autouse=True)
+def setup_teardown():
+    storage_client = storage.Client()
+    bucket = storage_client.create_bucket(STAGING_BUCKET)
+    bucket.blob(f"{INPUT_DIR}/{TRAIN_DATA}").upload_from_filename(TRAIN_DATA, timeout=600)
+    
+    with tarfile.open(TRAINER_TAR, mode="x:gz") as tar:
+        tar.add(f"{TRAINER_DIR}/")
+
+    bucket.blob(TRAINER_TAR).upload_from_filename(TRAINER_TAR)
+
+    yield bucket
+
+    [blob.delete() for blob in bucket.list_blobs()]
+    bucket.delete()
+
+    os.remove(TRAINER_TAR)
+
+@pytest.mark.timeout(1200)
+def test_sklearn(setup_teardown):
+    ml = discovery.build('ml', 'v1')
```

Comment:
Instead of the discovery library, this should migrated to use uCAIP (and later Model Builder SDK) at some point.

Reply:
Let me know if you need help on this.

Reply:
There's a lot of other components of this tutorial, such as composer operators, that don't currently work with uCAIP. We want to launch this as-is and hold off from migrating the whole thing to uCAIP for now.

Reply:
Make sense!

Reply:
Update: migrated the sample!

Reply:
Thanks!